scripts/dotnet-cli-build/PrepareTargets.cs

```diff
@@ -200,6 +172,162 @@ public static BuildTargetResult RestorePackages(BuildTargetContext c)
             return c.Success();
         }
 
+        [Target]
+        public static BuildTargetResult CheckPrereqDebianPackageBuildComponents(BuildTargetContext c)
+        {
+            if (!CurrentPlatform.IsUbuntu)
+            {
+                return c.Success();
+            }
+
+            var debianPackageBuildDependencies = new string[]
+            {
+                "devscripts", 
+                "debhelper",
+                "build-essential"
+            };
+
+            var messageBuilder = new StringBuilder();
+
+            foreach (var package in debianPackageBuildDependencies)
+            {
+                if (!AptPackageIsInstalled(package))
+                {
+                    messageBuilder.Append($"Error: Debian package build dependency {package} missing.");
+                    messageBuilder.Append(Environment.NewLine);
+                    messageBuilder.Append($"-> install with apt-get install {package}");
+                    messageBuilder.Append(Environment.NewLine);
+                }
+            }
+
+            if (messageBuilder.Length == 0)
+            {
+                return c.Success();
+            }
+            else
+            {
+                return c.Failed(messageBuilder.ToString());
+            }
+        }
+
+        [Target]
+        public static BuildTargetResult CheckPrereqCoreclrDependencyPackages(BuildTargetContext c)
+        {
+            if (!CurrentPlatform.IsUbuntu && !CurrentPlatform.IsCentOS)
```

Comment:
We can also do something like xunit (example -https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets.Legacy/tests/FunctionalTests/SendPacketsAsync.cs#L57)

Something like 

```
[Target(Platform.Linux)] // this will execute attributed method only on linux
```